stock_location_orderpoint/models/stock_move.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2023 Michael Tietz (MT Software) <mtietz@mt-software.de>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, fields, models
+from odoo.tools import ormcache
+
+from odoo.addons.queue_job.job import identity_exact
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    location_orderpoint_id = fields.Many2one(
+        "stock.location.orderpoint", "Stock location orderpoint"
+    )
+
+    @ormcache("self", "product")
+    def _get_location_orderpoint_replenishment_date(self, product):
+        return min(
+            self.filtered(lambda move: move.product_id == product).mapped("date")
+        )
+
+    def _prepare_auto_replenishment_for_waiting_moves(self):
+        self._prepare_auto_replenishment(
+            "location_id",
+            self.env["stock.location.orderpoint"]._get_waiting_move_domain(),
+        )
+
+    def _prepare_auto_replenishment_for_done_moves(self):
+        self._prepare_auto_replenishment(
+            "location_dest_id",
+            [
+                ("move_dest_ids", "=", False),
+                ("procure_method", "=", "make_to_stock"),
+                ("state", "=", "done"),
+            ],
+        )
+
+    def _prepare_auto_replenishment(self, location_field, domain):
+        if self.env.context.get("skip_auto_replenishment"):
+            return
+        locations_products = {}
+        product_obj = self.env["product.product"]
+        for move in self.filtered_domain(domain):
+            location = getattr(move, location_field)
+            locations_products.setdefault(location, set())
+            locations_products[location].add(move.product_id.id)
+        # Map the the move's location field
+        # to the correspoding stock.location.orderpoint's location field
+        location_field = (
+            location_field == "location_id" and location_field or "location_src_id"
+        )
+        for location, products in locations_products.items():
+            orderpoints = self.env["stock.location.orderpoint"]._get_orderpoints(
```

Comment:
This will generate 1 SQL qry by loop. The method should be called from outside the loop and called only once for all the locations....

Reply:
If we calling it outside of the loop, then we have to do child_of check inside the loop, because the _get_orderpoints method is searching with parent_of.

Reply:
@mt-software-de `child_of'check inside the loop is a simple python method `child_location.parent_path.startswith(parent_location.parent_path)`  It's far less expensive than an SQL query by loop...